### ISSUE TITLE
Is it possible to add support for GenerateCreateScript?

### ISSUE THREAD

**Issue by stevebelskie** [rr-wfm/MSBuild.Sdk.SqlProj#27]
It seems that something in the vein of the example below could be added to `PackageBuilder` or in `Extensions`, that could be called based on a new property added to `PackageOptions`.
```cs
public void GenerateCreateScript(FileInfo dapacFile)
{
  using (var package = DacPackage.Load(dap.FullName))
  using(var file = File.Create("some-parameterized-name.sql"))
  {
     DacServices.GenerateCreateScript(file, package, "test");
               
  }
}
```
Not super familiar with the underlying apis here so there may be a better way using than `DacServices` or a reason that it's not feasible for this project. 

In any case, thanks for this awesome project!

**Comment by jmezach** [comment]
Hi @stevebelskie. Thanks for the suggestion. Would you mind elaborating a little bit about what you are trying to achieve? So far this project has focussed on being able to produce a `.dacpac` from a set of SQL files, while leaving the deployment to existing tools in the ecosystem (namely `SqlPackage.exe`). I wouldn't mind adding an option to generate a `.sql` script as well, but I'm struggling to find a real use case for it.

**Comment by stevebelskie** [comment]
So it's an option available in a `.sqlproj` that is distinct from the option to generate an script when publishing/deploying. The latter I believe is what `SqlPackage.exe` provides. I couldn't figure out a away to generate a script using `SqlPackage.exe` without specifying an actual target db. The `GenerateCreateScript` option (as far as I know) simply creates a sql script that creates the entire database. So as far as I understand it, it creates the `.dapac` in the form of a sql script, it doesn't generate an actual deployment script for some target. 

I'm mostly exploring the idea of using `.dapacs` in furture projects, so this is only somewhat thoughout. The use case for me would be for integration tests run in a docker container. I generally make use of `sqlcmd` to setup an in container db and seed data. It would be nice to be able to run the sql script in those cases so as to avoid having to install `SqlPackage.exe` (the `mssql/server:2019-latest` image comes with `sqlcmd` already) or worry about `.dapac` deployment at all given that I'm just creating and destroying the database in one go.

**Comment by jmezach** [comment]
That makes perfect sense actually. So would you like to have both the script and the .dacpac? I'm not entirely familiar with how this works with the .sqlproj format since I haven't used that feature. Could you briefly explain how it works there, or point me to some doc that explains how it works? Ideally I would like to keep things the same as much as I can.

**Comment by stevebelskie** [comment]
Yes, you end up with both the .dacpac and the script. I'm not able to find anything as far as documentation that the the feature exists for .sqlproj. I only stumbled upon it as an option in Visual Studio:
![image](https://user-images.githubusercontent.com/37506132/85933103-3401cc80-b8a1-11ea-9bb5-95c777780a52.png)
It adds an xml element to the main property group: `<GenerateCreateScript>True</GenerateCreateScript>` and that seems to be it. I'm not aware of any other options or interactions with other properties that would affect the behavior. 

Googling didn't bring up much other than the API page for method in the example above. ([link](https://docs.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.dac.dacservices.generatecreatescript?view=sql-dacfx-140.3881.1))

**Comment by jmezach** [comment]
Okay, cool, that doesn't look too hard to add. We'll just need to add a parameter to the command line tool to toggle this behaviour and then pass the value of that property to it. 

I'll see if I can find the time to add this. A PR would be much appreciated though.

**Comment by stevebelskie** [comment]
I'd be happy to contribute a PR. I should have time this week to do so.

**Comment by jmezach** [comment]
That would be awesome!

**Comment by jmezach** [comment]
@stevebelskie Is this something you're still working on?

**Comment by stevebelskie** [comment]
@jmezach hey there, been a busy few months, but still on my radar. I've had it mostly finished for a bit now. I can probably wrap it up this weekend.

**Comment by jmezach** [comment]
@stevebelskie Just a heads-up that I've merged #33 earlier today which introduced some significant changes to the codebase due to some renaming. You'll probably want to merge with master.

**Comment by JonDBA** [comment]
I came here looking for the same thing. I've only had a chance to get a high level look at the code but here's what I'm seeing:

Layering [DacServices.Publish](https://docs.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.dac.dacservices.publish) on top of most of the existing [DacServices.Deploy](https://docs.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.dac.dacservices.deploy) logic should provide the desired result.

Using the existing [v1.2.0](https://github.com/jmezach/MSBuild.Sdk.SqlProj/tree/v1.2.0) implementation of [DacDeployOptions](https://docs.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.dac.dacdeployoptions) as part of the [PublishOptions](https://docs.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.dac.publishoptions) and exposing [PublishOptions.DatabaseScriptPath](https://docs.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.dac.publishoptions.databasescriptpath) and [PublishOptions.GenerateDeploymentScript](https://docs.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.dac.publishoptions.generatedeploymentscript) as publish options seems promising.

If no-one else is actively working on this I may be able to hack something together and submit a PR.

**Comment by jmezach** [comment]
I'm not entirely sure if adding this to the publishing pipeline. Personally I feel like the script would be more of a build artifact, rather than a publish artifact. Not sure how SSDT does it though.

**Comment by ErikEJ** [comment]
This issue is about an additional build artifact, besides the .dacpac file, so definitely nothing to do with deployment.

**Comment by jmezach** [comment]
@stevebelskie @JonDBA Is either one of you actively working on this? No worries if not, just looking for a status update ;).

**Comment by stevebelskie** [comment]
@jmezach I haven't been able to find the time recently to tackle this, so anyone else should feel free. Otherwise, I'll try to revisit next month if I can find the time.

**Comment by jmezach** [comment]
No problem, we aren't in a rush here. Take all the time you need. In the mean time if anyone else wants to implement this please feel free to do so and let us know here.

**Comment by joshsten** [comment]
Came here looking for this for the exact same reason (integration tests.)

**Comment by markalroberts** [comment]
And me :)

**Comment by jmezach** [comment]
This has been implemented in 1.15.0 thanks to @markalroberts.
